File: lab-3-ddb/environment/node_8.10.0/query_table.js
var 
    AWS = require("aws-sdk"),                            
    DDB = new AWS.DynamoDB({
        apiVersion: "2012-08-10",
        region: "<FMI>"
    });                                                   

(function queryTable(){
    var 
        params = {
            <FMI>: {
                ":catname": {
                    <FMI>: "<FMI>"
                }
            },
            KeyConditionExpression: "petname = :catname",
            <FMI>: "notable_features",
            TableName: "lostcats"
        };
    DDB.<FMI>(params, function(err, data){
        if(err){
            throw err;
        }
        // TIP: notable_features is a String
        console.log(data.Items[0].notable_features.<FMI>);  
    });
})();

File: lab-3-ddb/environment/ruby_2.6.0/query_table.rb
require "aws-sdk-dynamodb"

dynamodb_client = Aws::DynamoDB::Client.new(region: "<FMI>")

table_name = "lostcats"
name = "Puddles"

params = {
    table_name: table_name,
    expression_attribute_values: {
        ":p" => name,
    },
    key_condition_expression: "<FMI>",
}

begin
    result = <FMI>(params)
    puts "Query succeeded."
    result.items.each{|petname|
         puts "#{petname["<FMI>"]}"
    }

rescue  Aws::DynamoDB::Errors::ServiceError => error
    puts "Unable to query table:"
    puts "#{error.message}"
end